workflows-cli/src/linter/mod.rs

```diff
@@ -0,0 +1,212 @@
+#![allow(clippy::missing_docs_in_private_items)]
+
+mod base_linting;
+
+use base_linting::lint_from_manifest;
+mod helm;
+use helm::lint_from_helm;
+mod config_linting;
+
+use clap::ValueEnum;
+
+use crate::{LintArgs, LintConfigArgs, linter::config_linting::lint_from_config};
+
+/// Supported manifest
+#[derive(Debug, Clone, ValueEnum)]
+#[clap(rename_all = "lower")]
+pub enum ManifestType {
+    Manifest,
+    Helm,
+}
+
+#[derive(Debug, PartialEq, PartialOrd, Eq, Ord)]
+struct LintResult {
+    name: String,
+    errors: Vec<String>,
+}
+
+impl LintResult {
+    fn new(name: String, errors: Vec<String>) -> Self {
+        Self { name, errors }
+    }
+}
+
+pub fn config_lint(args: LintConfigArgs) {
+    lint_from_config(&args.config_file);
+}
+
+/// Linter entrypoint
+pub fn lint(args: LintArgs) {
+    let results = match &args.manifest_type {
+        ManifestType::Manifest => lint_from_manifest(&args.file_name, args.all),
+        ManifestType::Helm => lint_from_helm(&args.file_name, args.all),
+    };
+
+    match results {
+        Ok(results) => print_and_exit(results),
+        Err(e) => println!("Something went wrong while linting:\n{}", e),
+    }
+}
+
+fn print_and_exit(results: Vec<LintResult>) {
+    results.iter().for_each(|result| {
+        print!("Template Name: {} - ", result.name);
+
+        if result.errors.is_empty() {
+            println!("No errors");
+        } else {
+            println!("Errors found");
+            result.errors.iter().for_each(|e| println!("  - {}", e));
+        }
+    });
+    match results.iter().any(|result| !result.errors.is_empty()) {
+        true => std::process::exit(1),
+        false => std::process::exit(0),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::env;
+    use std::path::Path;
+
+    use serial_test::serial;
+
+    use crate::linter::{LintResult, base_linting::lint_from_manifest, helm::lint_from_helm};
+
+    fn have_same_elements<T: Ord>(result: &mut Vec<T>, expected: &mut Vec<T>) -> bool {
+        result.sort();
+        expected.sort();
+        result == expected
+    }
+
+    #[test]
+    #[serial]
+    fn lint_single_passing_manifest() {
+        unsafe {
+            env::set_var("RUST_TEST_ACTIVE_MAPPING", "passing_manifest");
```

Comment:
Why did you choose to modify the behavior of the MockCommand via environment variables? This seems to me that it might be potentially confusing to a future maintainer.

As a minimum, I would suggest adding comments to explain to a future maintainer what these environment variables are doing.